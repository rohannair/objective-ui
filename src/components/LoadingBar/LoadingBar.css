@import '../../styles/globals';
@import '../../styles/mixins';
@import '../../styles/layout';

.LoadingBar {
  @extend container;
}

.loading {
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
}
