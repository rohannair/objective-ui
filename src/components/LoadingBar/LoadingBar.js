import React, { PropTypes } from 'react';
import styles from './LoadingBar.css';
import classNames from 'classnames/bind';

import ProgressBar from 'react-toolbox/lib/progress_bar';

let cx = classNames.bind(styles);

const LoadingBar = p => {
  const classname = cx({
    [styles.LoadingBar]: true
  });

  return (
    <div className={classname}>
      <div className={styles.loading}>
        <ProgressBar multicolor type="circular" mode="indeterminate" />
      </div>
    </div>
  );
};

export default LoadingBar;
