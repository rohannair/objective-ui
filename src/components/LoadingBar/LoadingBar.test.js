import React from 'react'
import LoadingBar from './index.js'

describe('Test suite for LoadingBar component', () => {
  it('LoadingBar should exist', () => {
    let wrapper = shallow(<LoadingBar/>)
    expect(wrapper).toMatchSnapshot()
  })
})

