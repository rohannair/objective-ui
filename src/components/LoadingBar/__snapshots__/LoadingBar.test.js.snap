exports[`Test suite for LoadingBar component LoadingBar should exist 1`] = `
<div
  className="undefined">
  <div>
    <ThemedProgressBar
      buffer={0}
      className=""
      composeTheme="deeply"
      max={100}
      min={0}
      mode="indeterminate"
      multicolor={true}
      type="circular"
      value={0} />
  </div>
</div>
`;
